test/parallel_api/algorithm/alg.sorting/alg.set.operations/set.pass.h

```diff
@@ -0,0 +1,365 @@
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "support/test_config.h"
+
+#include _PSTL_TEST_HEADER(execution)
+#include _PSTL_TEST_HEADER(algorithm)
+
+#include "support/utils.h"
+
+#include <cmath>
+#include <chrono>
+
+#if  !defined(_PSTL_TEST_SET_UNION) && !defined(_PSTL_TEST_SET_DIFFERENCE) && !defined(_PSTL_TEST_SET_INTERSECTION) &&\
+     !defined(_PSTL_TEST_SET_SYMMETRIC_DIFFERENCE)
+#define _PSTL_TEST_SET_UNION
+#define _PSTL_TEST_SET_DIFFERENCE
+#define _PSTL_TEST_SET_INTERSECTION
+#define _PSTL_TEST_SET_SYMMETRIC_DIFFERENCE
+#endif
```

Comment:
These macros are not required anymore, because there are separated files for each algorithm. 
Some time ago, they were added to build a test for a specific algorithm to determine an algorithm-related pass rate.

Could you remove them?

Reply:
No: this macro still required in each test and simple moved from original file to new header file.

Reply:
I see that removing the macros from `set_common.h` is not straightforward, because they are now used in `test_set`.

Meanwhile, removing them from the newly created files, which are intended for a certain algorithm, is easy and makes sense (to me at least). There is no any application in building a test with a macro which turns it off entirely. It used to be useful when all the algorithms were tested in a single file - to filter algorithms to test.

You may also consider splitting `test_usm` function, so that it tests certain algorithm in a designated file. For example, `set_union.pass.cpp` has all `set_union` cases and so on. Then, `set.pass.cpp` would be not required. This is optional.

Reply:
It depends on a situation. Sometimes it is reasonable and sometimes it is not.

Some tests are entirely skipped when a certain backend is not available. For example, `buffer_wrapper.pass.cpp` is skipped when dpcpp backend is not available. This is done on purpose.

Having a macro e.g. `_PSTL_TEST_SET_UNION` to skip the whole `set_union` test is not applicable... These macros became not relevant after splitting a test for each algorithm. That is why I suggest removing them, at least from files designated for a specific algorithm.

Reply:
My proposal to do this in separate PR.